### PR TITLE
Bump CRT version and expose setting memory limits for S3 calls

### PR DESCRIPTION
**Pull request by cenedhryn**
## Motivation and Context
CRT has published a new setting to limit memory used when making S3 calls.

## Modifications
- Updated CRT version
- Added new parameter
- Added Javadoc description for parameter, and for throughput

## Testing
Unit tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [x] My change requires a change to the Javadoc documentation
- [x] I have updated the Javadoc documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: services/s3/src/main/java/software/amazon/awssdk/services/s3/S3CrtAsyncClientBuilder.java [hunk 1]
```diff
@@ -128,10 +146,14 @@ default S3CrtAsyncClientBuilder credentialsProvider(IdentityProvider<? extends A
      * instance type in <a href="https://aws.amazon.com/ec2/instance-types/">Amazon EC2 instance type page</a>.
      * If you are running into out of file descriptors error, consider using {@link #maxConcurrency(Integer)} to limit the
      * number of connections.
+     * <p>
+     * <b>Note: </b> this setting affects the memory usage of CRT; a higher throughput value will result in a larger memory
```
Comment (zoewangg) [inline services/s3/src/main/java/software/amazon/awssdk/services/s3/S3CrtAsyncClientBuilder.java:150]: Suggestion: `...affects the native memory usage of CRT...`

File: services/s3/src/main/java/software/amazon/awssdk/services/s3/S3CrtAsyncClientBuilder.java [hunk 2]
```diff
@@ -114,12 +114,30 @@ default S3CrtAsyncClientBuilder credentialsProvider(IdentityProvider<? extends A
...
+     * <ul>
+     *     <li><b>Min: </b>1 GB</li>
+     *     <li><b>Max: </b>The lowest value of the supplied value and the SIZE_MAX of the system</li>
+     * </ul>
+     *
+     * @param memoryLimitInBytes the memory limit in bytes
+     * @return this builder for method chaining.
+     * @see #targetThroughputInGbps(Double)
+     */
+    S3CrtAsyncClientBuilder memoryLimitInBytes(Long memoryLimitInBytes);
```
Comment (zoewangg) [inline services/s3/src/main/java/software/amazon/awssdk/services/s3/S3CrtAsyncClientBuilder.java:133]: Suggestion `The amount of native memory that ...`. How about `maxNativeMemoryUsageInBytes` or `maxMemoryUsageInBytes`?

File: .changes/next-release/feature-AWSCRTbasedS3Client-353870f.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+    "type": "feature",
+    "category": "AWS CRT-based S3 Client",
```
Comment (dagnir) [inline .changes/next-release/feature-AWSCRTbasedS3Client-353870f.json:3]: Technically affects other CRT related components like signing right? Might be better just as "AWS SDK for Java v2"

**Review by dagnir — APPROVED** [review]

File: .changes/next-release/feature-AWSCRTbasedS3Client-353870f.json (hunk 3 above)
Comment (cenedhryn) [reply .changes/next-release/feature-AWSCRTbasedS3Client-353870f.json:3]: I looked at other entries in the changelog, but perhaps we're not consistent - not surprising.

Comment (zoewangg) [reply .changes/next-release/feature-AWSCRTbasedS3Client-353870f.json:3]: Hmm, I thought it only affects s3 client, is that not the case?

Comment (zoewangg) [reply .changes/next-release/feature-AWSCRTbasedS3Client-353870f.json:3]: oh nvm, this is version bump, yeah, should be `AWS SDK for Java v2`

File: .changes/next-release/feature-AmazonS3-c28f9de.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+    "type": "feature",
+    "category": "Amazon S3",
```
Comment (zoewangg) [inline .changes/next-release/feature-AmazonS3-c28f9de.json:3]: Should this be `AWS CRT-based S3 Client`?

**Review by zoewangg — APPROVED** [review]

**Pull request merged**